packages/common/src/components/PkgGraph.tsx

```diff
@@ -0,0 +1,193 @@
+import {
+  MainAreaWidget,
+  ReactWidget
+} from '@jupyterlab/apputils';
+
+import { Widget } from '@lumino/widgets';
+
+import { HTMLSelect } from '@jupyterlab/ui-components';
+
+import cytoscape from 'cytoscape';
+
+import * as React from 'react';
+
+import { CondaEnvironments } from '../services';
+import { Conda } from '../tokens';
+
+class Graph extends Widget {
+  constructor(model: CondaEnvironments) {
+    super();
+    this._model = model;
+    this._selected = '';
+    this._pkgManager = this._model.getPackageManager();
+    this._updatePackages();
+  }
+
+  update(): void {
+    const layout = (): cytoscape.LayoutOptions => {
+      const ns = new Set<string>();
+      this._nodes.forEach((n: cytoscape.NodeDefinition) => {
+        ns.add(n.data.name);
+      });
+      if (ns.size >= 50) {
+        return {
+          name: 'circle',
```

Comment:
Looking at the previous implementation from: https://github.com/mamba-org/mamba-navigator/blob/84a3d5a700b30017112d68b98fdd7e63adc9f182/src/components/Network.vue#L70

It looks like it was using the `vis.Network`, which is likely to be the one from https://visjs.org:

![image](https://user-images.githubusercontent.com/591645/98537141-19952200-2289-11eb-9a3e-c3e896a5b0fd.png)

Maybe it's a just a matter of using an equivalent as a cytoscape plugin to have a similar layout?

Reply:
Yes, this one looks nicer.
I'll look for a similar layout and modify the backend to get the dependencies.

Maybe something like this one: https://cytoscape.org/cytoscape.js-cose-bilkent/

Reply:
it could also be good to have a lighter library than cytoscape.

Possible alternative: 
* https://danielcaldas.github.io/react-d3-graph/docs/index.html
* vis.js has been splitted; the network part is there https://github.com/visjs/vis-network (react wrapper: https://github.com/crubier/react-graph-vis)

Reply:
Thanks both for the info.
I changed the library and added an argument to the `PackageHandler` API, so now we can request package dependencies.

![Screen-Recording-2020-11-16-at-13 45 22](https://user-images.githubusercontent.com/26092748/99254861-53c76c00-2813-11eb-9047-d66a38fa5af6.gif)

Would be great to access this graph when clicking on a package in `jupyter_conda`. Maybe having a new column for dependencies or a link in the package description to open the graph in a `dialog`.

Let me know what do you think.

Reply:
Looks great!

> Maybe having a new column for dependencies or a link in the package description to open the graph in a dialog.

Yes maybe that's the simplest and most visible way for now?